src/store/types.ts

```diff
@@ -0,0 +1,89 @@
+import { Willow } from "../../deps.ts";
+import { AuthorisationToken } from "../auth/auth.ts";
+import { IdentityAddress, ShareAddress } from "../crypto/types.ts";
+import { Base32String } from "../encoding/types.ts";
+
+export type Path = string[];
+
+export type Document = {
+  share: ShareAddress;
+  identity: IdentityAddress;
+  path: string[];
+  timestamp: bigint;
+  size: bigint;
+  digest: Base32String;
+  signedBy: IdentityAddress;
+  payload: Willow.Payload | undefined;
+};
+
+export type StoreDriverOpts = "memory" | {
+  entryDriver: Willow.EntryDriver<
+    ShareAddress,
+    IdentityAddress,
+    ArrayBuffer,
+    ArrayBuffer
+  >;
+  payloadDriver: Willow.PayloadDriver<ArrayBuffer>;
+};
+
+export type Query = {
+  /** A path all documents must be prefixed by. */
+  pathPrefix?: string[];
```

Comment:
```suggestion
  pathPrefix?: Path;
```

Semantically should this be a Path here?

Reply:
Yes, it should! Thanks for catching that.